### PR TITLE
[SYCL] Fix SYCL_DEVICE_FILTER setting

### PR DESCRIPTION
**Pull request by bso-intel**
HOST device is not automatically available when SYCL_DEVICE_FILTER is set.
SYCL_DEVICE_FILTER has to list 'host' explicitly if a test needs it.

Signed-off-by: Byoungro So <byoungro.so@intel.com>

### CODE REVIEW

File: SYCL/FilterSelector/select_device_cpu.cpp [hunk 1]
```diff
@@ -45,12 +45,19 @@ int main() {
     device d = cs.select_device();
     std::cout << "CPU device is found: " << d.is_cpu() << std::endl;
   }
-  // HOST device is always available regardless of SYCL_DEVICE_FILTER
+  /*
+  // TODO: enable this once SYCL_DEVICE_FILTER PR is merged.
```
Comment (bso-intel) [inline SYCL/FilterSelector/select_device_cpu.cpp:49]: https://github.com/intel/llvm/pull/3397 needs to be merged first to enable this test.
PR-3397 forces SYCL_DEVICE_FILTER to only provide designated devices.

**Review by alexbatashev — APPROVED** [review]
LGTM

**Comment by bso-intel** [comment]
@alexbatashev Thanks.
@vladimirlaz Could you merge this so I can make progress in my SYCL RT PR?
Thanks.

**Comment by bso-intel** [comment]
@vladimirlaz Could you merge this? Thanks.

**Pull request merged**